File: admin-backend/application-api/src/main/java/com/xfrog/platform/application/base/DicApi.java
package com.xfrog.platform.application.base;

import com.xfrog.framework.dto.PageDTO;
import com.xfrog.platform.application.base.dto.CreateDicItemRequestDTO;
import com.xfrog.platform.application.base.dto.CreateDicRequestDTO;
import com.xfrog.platform.application.base.dto.DicDTO;
import com.xfrog.platform.application.base.dto.QueryDicRequestDTO;
import com.xfrog.platform.application.base.dto.UpdateDicItemRequestDTO;
import com.xfrog.platform.application.base.dto.UpdateDicRequestDTO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.List;

@Tag(name = "DicApi", description = "字典管理接口")
@RequestMapping("/api/dics")
public interface DicApi {
    @PostMapping
    @Operation(summary = "创建字典")
    Long createDic(@Valid @RequestBody CreateDicRequestDTO dic);

    @PostMapping("/list")
    @Operation(summary = "查询字典列表")
    PageDTO<DicDTO> listDics(@Valid @RequestBody QueryDicRequestDTO queryDTO);

    @GetMapping("/{dicId}")
    @Operation(summary = "查询字典")
    DicDTO getDic(@PathVariable("dicId") Long dicId);

    @PostMapping("/list/by-types")
    @Operation(summary = "根据类型查询字典")
    List<DicDTO> getDicByTypes(@RequestBody List<String> dicTypes);

    @PostMapping("/{dicId}")
    @Operation(summary = "更新字典")
    void updateDic(@PathVariable("dicId") Long dicId, @Valid @RequestBody UpdateDicRequestDTO dic);

    @DeleteMapping("/{dicId}")
    @Operation(summary = "删除字典")
    void deleteDic(@PathVariable("dicId") Long dicId);

    @PostMapping("/{dicId}/items")
    @Operation(summary = "创建字典项")
    Long createDicItem(@PathVariable("dicId") Long dicId, @Valid @RequestBody CreateDicItemRequestDTO requestDTO);

    @PutMapping("/{dicId}/items/{itemId}")
    @Operation(summary = "更新字典项")
    void updateDicItem(@PathVariable("dicId") Long dicId, @PathVariable("itemId") Long itemId, @Valid @RequestBody UpdateDicItemRequestDTO requestDTO);

}

File: admin-ui/src/services/swagger/dicApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request';

/** 创建字典 POST /api/dics */
export async function createDic(body: API.CreateDicRequestDTO, options?: { [key: string]: any }) {
  return request<number>('/api/dics', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 查询字典 GET /api/dics/${param0} */
export async function getDic(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getDicParams,
  options?: { [key: string]: any },
) {
  const { dicId: param0, ...queryParams } = params;
  return request<API.DicDTO>(`/api/dics/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 更新字典 POST /api/dics/${param0} */
export async function updateDic(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateDicParams,
  body: API.UpdateDicRequestDTO,
  options?: { [key: string]: any },
) {
  const { dicId: param0, ...queryParams } = params;
  return request<any>(`/api/dics/${param0}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 删除字典 DELETE /api/dics/${param0} */
export async function deleteDic(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.deleteDicParams,
  options?: { [key: string]: any },
) {
  const { dicId: param0, ...queryParams } = params;
  return request<any>(`/api/dics/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 创建字典项 POST /api/dics/${param0}/items */
export async function createDicItem(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.createDicItemParams,
  body: API.CreateDicItemRequestDTO,
  options?: { [key: string]: any },
) {
  const { dicId: param0, ...queryParams } = params;
  return request<number>(`/api/dics/${param0}/items`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 更新字典项 PUT /api/dics/${param0}/items/${param1} */
export async function updateDicItem(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateDicItemParams,
  body: API.UpdateDicItemRequestDTO,
  options?: { [key: string]: any },
) {
  const { dicId: param0, itemId: param1, ...queryParams } = params;
  return request<any>(`/api/dics/${param0}/items/${param1}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 查询字典列表 POST /api/dics/list */
export async function listDics(body: API.QueryDicRequestDTO, options?: { [key: string]: any }) {
  return request<API.PageDTODicDTO>('/api/dics/list', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}
